dotCMS/src/integration-test/java/com/dotmarketing/db/SystemEnvDataSourceStrategyTest.java

```diff
@@ -0,0 +1,86 @@
+package com.dotmarketing.db;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotcms.util.TestInitialContext;
+import com.dotmarketing.util.Constants;
+import com.liferay.util.SystemEnvironmentProperties;
+import com.zaxxer.hikari.HikariDataSource;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.naming.NamingException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * Tests for {@link SystemEnvDataSourceStrategy}
+ * @author nollymar
+ */
+public class SystemEnvDataSourceStrategyTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Test
+    public void testApply() throws SQLException, NamingException {
+        final Map<String, String> properties = new HashMap<>();
+        final HikariDataSource testDatasource = (HikariDataSource) TestInitialContext.getInstance()
+                .getDataSource();
+        properties.put("connection_db_driver", testDatasource.getDriverClassName());
+        properties.put("connection_db_base_url", testDatasource.getJdbcUrl());
+        properties.put("connection_db_username", testDatasource.getUsername());
+        properties.put("connection_db_password", testDatasource.getPassword());
+        properties.put("connection_db_max_total", "60");
+        properties.put("connection_db_max_idle", "10");
+        properties.put("connection_db_max_wait", "60000");
+        properties.put("connection_db_validation_query", "SELECT 1");
+        properties.put("connection_db_leak_detection_threshold", "60000");
+
+        final SystemEnvironmentProperties systemEnvironmentProperties = Mockito.mock(SystemEnvironmentProperties.class);
+
+        properties.forEach((k,v) -> Mockito.when(systemEnvironmentProperties.getVariable(k)).thenReturn(v));
+
+        SystemEnvDataSourceStrategy strategy = new SystemEnvDataSourceStrategy(systemEnvironmentProperties);
+
+        final HikariDataSource dataSource = (HikariDataSource)
+                strategy.apply();
+
+        assertNotNull(dataSource);
+        assertNotNull(dataSource.getConnection());
+        validateConfiguration(dataSource, properties);
+    }
+
+    /**
+     * Verifies all values in the dataSource object are set correctly
+     * @param dataSource {@link HikariDataSource} to be validated
+     * @param properties {@link Map} contains the expected values to be compared
+     */
+    private void validateConfiguration(final HikariDataSource dataSource, Map<String, String> properties) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'properties' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5026767)